### PR TITLE
Add separate optgroups for application status filtering for program admin.

### LINKED ISSUE #3269: Status Tracking: Allow filters for the Application list
The entire Application List view is being revamped so we'll need to do a low-waste take at it in the current UI.

A MVP should allow for a filter selector showing
- all options available in the Applications that could be listed
- An option to show those with no status that can have a status selected.

Note: For applications on programs without statuses there is no status option available.

### PR DESCRIPTION
**Pull request by clouser2**
### Description

This allows for visual separation of different application status filters (e.g. "General" filters such as all applications) vs for a specific status.

## Release notes:

Better organize application status filter options within the program admin view.

### Checklist

- [X] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >
- [X] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)

Fixes #3269

### CODE REVIEW

File: server/app/views/components/SelectWithLabel.java [hunk 1]
```diff
@@ -93,21 +102,49 @@ public DivTag getSelectTag() {
...
-                        .withValue(optionData.value())
-                        .withCondSelected(optionData.value().equals(fieldValue));
+                  optionGroup -> {
+                    return optgroup()
+                        .withLabel(optionGroup.label())
+                        .with(
+                            each(
+                                optionGroup.options(),
+                                optionData -> {
+                                  return option(optionData.label())
```
Comment (shanemc-goog) [inline server/app/views/components/SelectWithLabel.java:114]: Consider pulling this each computation out for readability, it'd remove a lot of nesting and a lambda return.

File: server/test/views/components/SelectWithLabelTest.java [hunk 2]
```diff
@@ -45,5 +45,27 @@ public void createSelect_rendersSelectedOption() {
...
+
+  @Test
+  public void createSelect_withOptGroups() {
+    SelectWithLabel selectWithLabel =
+        new SelectWithLabel()
+            .setId("id")
+            .setOptionGroups(
+                ImmutableList.of(
+                    SelectWithLabel.OptionGroup.builder()
+                        .setLabel("A label")
```
Comment (shanemc-goog) [inline server/test/views/components/SelectWithLabelTest.java:59]: What about 2 groups??!?!?!?! :)

**Review by shanemc-goog — APPROVED** [review]

File: server/app/views/components/SelectWithLabel.java (hunk 1 above)
Comment (clouser2) [reply server/app/views/components/SelectWithLabel.java]: I moved the computation of both the `<optgroup>` and `<option>` tags out into separate methods, which removed a fair amount of the nesting, which is what I think you're after.

File: server/test/views/components/SelectWithLabelTest.java (hunk 2 above)
Comment (clouser2) [reply server/test/views/components/SelectWithLabelTest.java]: Done.

**Review by clouser2 — COMMENTED** [review]
Feedback incorporated. Since there was an approval, I'll go ahead and merge and happy to address any post-merge feedback in a separate PR.

**Review by shanemc-goog — APPROVED** [review]

**Pull request merged**